src/utilily/firebaseTypes.ts

```diff
@@ -0,0 +1,18 @@
+import { User } from "./types";
+
+export type UserClames = {
+  admin?: boolean;
+  superadmin?: boolean;
+  user?: boolean;
+
+  aud?: string;
+  auth_time?: number;
+  email?: string;
+  email_verified?: boolean;
+  exp?: number;
+  firebase?: { identities: { email: [string] }; sign_in_provider: string };
+  iat?: number;
+  iss?: string;
+  sub?: string;
+  user_id?: User["id"];
```

Comment:
I'd say we only need our own custom claims here.

Reply:
We need to have type of the object we are receiving. Right now we are receiving claims object that looks like this
{
  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;
  aud?: string;
  auth_time?: number;
  email?: string;
  email_verified?: boolean;
  exp?: number;
  firebase?: { identities: { email: [string] }; sign_in_provider: string };
  iat?: number;
  iss?: string;
  sub?: string;
  user_id?: User["id"];
};  


I don't think it's a good idea to say that claims of type

 {  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;  }

And then press in it object that looks like this.

  {
  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;
  aud?: string;
  auth_time?: number;
  email?: string;
  email_verified?: boolean;
  exp?: number;
  firebase?: { identities: { email: [string] }; sign_in_provider: string };
  iat?: number;
  iss?: string;
  sub?: string;
  user_id?: User["id"];
};